File: app/src/main/java/org/grapheneos/pdfviewer/fragment/JumpToPageFragment.java
package org.grapheneos.pdfviewer.fragment;

import android.app.Dialog;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.NumberPicker;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import org.grapheneos.pdfviewer.PdfViewer;

public class JumpToPageFragment extends DialogFragment {
    public static final String TAG = "JumpToPageFragment";

    private final static String STATE_PICKER_CUR = "picker_cur";
    private final static String STATE_PICKER_MIN = "picker_min";
    private final static String STATE_PICKER_MAX = "picker_max";

    private NumberPicker mPicker;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (savedInstanceState != null) {
            mPicker.setMinValue(savedInstanceState.getInt(STATE_PICKER_MIN));
            mPicker.setMaxValue(savedInstanceState.getInt(STATE_PICKER_MAX));
            mPicker.setValue(savedInstanceState.getInt(STATE_PICKER_CUR));
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        if(getActivity()==null) {
            throw new RuntimeException("An unexpected error occurred. getActivity() was null in fragment DialogFragment.");
        }

        mPicker = new NumberPicker(getActivity());
        mPicker.setMinValue(1);
        mPicker.setMaxValue(((PdfViewer)getActivity()).mNumPages);
        mPicker.setValue(((PdfViewer)getActivity()).mPage);

        final FrameLayout layout = new FrameLayout(getActivity());
        layout.addView(mPicker, new FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.WRAP_CONTENT,
                FrameLayout.LayoutParams.WRAP_CONTENT,
                Gravity.CENTER));

        return new AlertDialog.Builder(getActivity())
                .setView(layout)
                .setPositiveButton(android.R.string.ok, (dialogInterface, i) -> {
                    mPicker.clearFocus();
                    ((PdfViewer)getActivity()).onJumpToPageInDocument(mPicker.getValue());
                })
                .setNegativeButton(android.R.string.cancel, null)
                .create();
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        outState.putInt(STATE_PICKER_MIN, mPicker.getMinValue());
        outState.putInt(STATE_PICKER_MAX, mPicker.getMaxValue());
        outState.putInt(STATE_PICKER_CUR, mPicker.getValue());
    }
}

File: app/src/main/java/org/grapheneos/pdfviewer/fragment/JumpToPageFragment.kt
package org.grapheneos.pdfviewer.fragment

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.Gravity
import android.widget.FrameLayout
import android.widget.NumberPicker
import androidx.fragment.app.DialogFragment
import org.grapheneos.pdfviewer.PdfViewer

private const val STATE_PICKER_CUR = "picker_cur"
private const val STATE_PICKER_MIN = "picker_min"
private const val STATE_PICKER_MAX = "picker_max"

class JumpToPageFragment : DialogFragment() {
    private lateinit var mPicker: NumberPicker

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        mPicker = NumberPicker(context).apply {
            minValue = savedInstanceState?.getInt(STATE_PICKER_MIN)
                ?: 1
            maxValue = savedInstanceState?.getInt(STATE_PICKER_MAX)
                ?: (activity as PdfViewer).mNumPages
            value = savedInstanceState?.getInt(STATE_PICKER_CUR)
                ?: (activity as PdfViewer).mPage
        }

        val layout = FrameLayout(requireContext())
        layout.addView(
            mPicker,
            FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.WRAP_CONTENT,
                FrameLayout.LayoutParams.WRAP_CONTENT,
                Gravity.CENTER
            )
        )

        return AlertDialog.Builder(context)
            .setView(layout)
            .setPositiveButton(android.R.string.ok) { _, _ ->
                mPicker.clearFocus()
                (activity as PdfViewer).onJumpToPageInDocument(mPicker.value)
            }
            .setNegativeButton(android.R.string.cancel, null)
            .create()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.apply {
            putInt(STATE_PICKER_MIN, mPicker.minValue)
            putInt(STATE_PICKER_MAX, mPicker.maxValue)
            putInt(STATE_PICKER_CUR, mPicker.value)
        }
    }

    companion object {
        const val TAG = "JumpToPageFragment"
    }
}